python/cudf/cudf/core/column/datetime.py

```diff
@@ -120,7 +120,8 @@ def normalize_binop_value(self, other):
             other = other.to_datetime64()
         elif isinstance(other, pd.Timedelta):
             other = other.to_timedelta64()
-
+        elif isinstance(other, cudf.Scalar):
+            return other
```

Comment:
Move this to the top of the conditions being checked to short circuit as much as possible.